subprojects/language-native/src/integTest/groovy/org/gradle/language/swift/SwiftApplicationIntegrationTest.groovy

```diff
@@ -467,39 +468,6 @@ class SwiftApplicationIntegrationTest extends AbstractSwiftIntegrationTest imple
         installation.assertIncludesLibraries("Greeter")
     }
 
-    def "fails when dependency library does not specify the same target machines"() {
```

Comment:
Why was this test removed?  Is it because it uses an unsupported architecture?  If it's the latter, we should just change this to use the opposite of the current os (for instance on linux, app -> linux, greeter -> macos).

Reply:
You are right. There is still a difference between unsupported architecture vs operating system. I change up this test and the C++ equivalent as well and added a specific test for the architecture on the C++ side.